src/workerd/server/server.c++

```diff
@@ -2558,15 +2552,14 @@ void startInspector(kj::StringPtr inspectorAddress,
     static constexpr uint DEFAULT_PORT = 9229;
 
     auto& network = io.provider->getNetwork();
-    auto inspectorListener = network.parseAddress(inspectorAddress, DEFAULT_PORT)
-      .then([](kj::Own<kj::NetworkAddress> parsed) {
-      return parsed->listen();
-    });
 
-    auto listen = inspectorListener.then([&inspectorService](kj::Own<kj::ConnectionReceiver> listener) mutable {
+    auto listen = (kj::coCapture([&network, &inspectorAddress, &inspectorService]()
```

Comment:
Pre-existing problem, so could be fixed in a separate PR, but: It appears that if this `listen` promise fails, nothing notices.

We should probably:
1. Replace the `kj::NEVER_DONE` below with `listen`.
2. Make the thread's lambda `noexcept`, so that if it throws, the whole process crashes.
3. Probably also throw if `listen` completes without an exception (though that shouldn't be possible).

Reply:
I'll add a TODO comment but would prefer to make that change separately.